cg/meta/archive/archive.py

```diff
@@ -84,8 +99,116 @@ def archive_all_non_archived_spring_files(
 
         for archive_location in ArchiveLocations:
             self.archive_archive_location(
-                files_and_samples=files_and_samples, archive_location=archive_location
+                files_and_samples=files_and_samples,
+                archive_location=archive_location,
+            )
+
+    def retrieve_flowcell(self, flowcell_name: str) -> None:
+        """Retrieves the archived spring files for all samples ran on the specified flowcell."""
+        flowcell: Flowcell = self.status_db.get_flow_cell_by_name(flowcell_name)
+        samples_to_fetch: List[SampleAndHousekeeperDestination] = self.get_samples_to_retrieve(
+            flowcell
+        )
+        for archive_location in ArchiveLocations:
+            filtered_samples: List[
+                SampleAndHousekeeperDestination
+            ] = filter_samples_on_archive_location(
+                samples_and_housekeeper_destinations=samples_to_fetch,
+                archive_location=archive_location,
+            )
+            if filtered_samples:
+                job_id: int = self.retrieve_samples_from_archive_location(
+                    samples_and_destinations=filtered_samples,
+                    archive_location=archive_location,
+                )
+                self.set_archive_retrieval_task_ids(
+                    retrieval_task_id=job_id,
+                    files=self.get_spring_files_from_samples(
+                        [sample.sample for sample in filtered_samples]
+                    ),
+                )
+
+    def get_spring_files_from_samples(self, samples: List[Sample]) -> List[File]:
```

Comment:
Could there be some files in the latest version which were not part of the retrieval? Or will setting the retrieval task id for all spring files in the latest version be a sufficient filtering?

Reply:
Changed to fetching only archived files and updating those.